### PR TITLE
Refactor dropdown to improve testing capabilities

### PR DESCRIPTION
**Pull request by raczyk**

### CODE REVIEW

File: src/components/dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -187,6 +197,7 @@ export const Dropdown: FunctionComponent<DropdownProps & SpaceProps> = ({
           {options &&
             options.map((option: string | IOption) => (
               <DropdownItem
+                title={typeof option === "string" ? option : option.value}
```
Comment (khola) [inline src/components/dropdown/Dropdown.tsx:200]: Why ?

File: src/components/dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -154,19 +162,21 @@ export const Dropdown: FunctionComponent<DropdownProps & SpaceProps> = ({
     <DropdownWrapper
       value={value}
       {...props}
-      onMouseLeave={() => isOpen && setIsOpen(!isOpen)}
+      // onMouseLeave={() => isOpen && setIsOpen(!isOpen)}
```
Comment (khola) [inline src/components/dropdown/Dropdown.tsx:165]: Why ?

**Review by khola — CHANGES_REQUESTED** [review]

File: src/components/dropdown/Dropdown.tsx (hunk 2 above)
Comment (raczyk) [reply src/components/dropdown/Dropdown.tsx:165]: I do forget to uncomment

File: src/components/dropdown/Dropdown.tsx (hunk 1 above)
Comment (raczyk) [reply src/components/dropdown/Dropdown.tsx:200]: For better queries in testing where you can query by value and compare it to the expected without matching key-value pairs

**Review by janosio-nordcloud — APPROVED** [review]

**Review by janosio-nordcloud — APPROVED** [review]

**Review by khola — APPROVED** [review]

**Pull request merged**